test_runner/src/main/kotlin/ftl/config/android/AndroidGcloudConfig.kt

```diff
@@ -0,0 +1,263 @@
+    )
+    @set:JsonProperty("test-runner-class")
+    var testRunnerClass: String? by data
+
+    @set:CommandLine.Option(
+        names = ["--test-targets"],
+        split = ",",
+        description = ["A list of one or more test target filters to apply " +
+                "(default: run all test targets). Each target filter must be fully qualified with the package name, class name, " +
+                "or test annotation desired. Any test filter supported by am instrument -e … is supported. " +
+                "See https://developer.android.com/reference/android/support/test/runner/AndroidJUnitRunner for more " +
+                "information."]
+    )
+    @set:JsonProperty("test-targets")
+    var testTargets: List<String?>? by data
+
+    @set:CommandLine.Option(
+        names = ["--robo-directives"],
+        split = ",",
+        description = [
+            "A comma-separated (<type>:<key>=<value>) map of robo_directives that you can use to customize the behavior of Robo test.",
+            "The type specifies the action type of the directive, which may take on values click, text or ignore.",
+            "If no type is provided, text will be used by default.",
+            "Each key should be the Android resource name of a target UI element and each value should be the text input for that element.",
+            "Values are only permitted for text type elements, so no value should be specified for click and ignore type elements."
+        ]
+    )
+    @set:JsonProperty("robo-directives")
+    var roboDirectives: Map<String, String>? by data
+
+    @set:CommandLine.Option(
+        names = ["--robo-script"],
+        description = [
+            "The path to a Robo Script JSON file.",
+            "The path may be in the local filesystem or in Google Cloud Storage using gs:// notation.",
+            "You can guide the Robo test to perform specific actions by recording a Robo Script in Android Studio and then specifying this argument.",
+            "Learn more at https://firebase.google.com/docs/test-lab/robo-ux-test#scripting. "
+        ]
+    )
+    @set:JsonProperty("robo-script")
+    var roboScript: String? by data
+
+    @CommandLine.Option(
+        names = ["--device"],
+        split = ",",
+        description = ["A list of DIMENSION=VALUE pairs which specify a target " +
+                "device to test against. This flag may be repeated to specify multiple devices. The four device dimensions are: " +
+                "model, version, locale, and orientation. If any dimensions are omitted, they will use a default value. Omitting " +
+                "all of the preceding dimension-related flags will run tests against a single device using defaults for all four " +
+                "device dimensions."]
+    )
+    fun deviceMap(map: Map<String, String>?) {
+        if (map.isNullOrEmpty()) return
+        val androidDevice = Device(
+            model = map.getOrDefault("model", FtlConstants.defaultAndroidModel),
```

Comment:
We use "model","version" etc in IosArgs and Android args. Maybe we should move keys to const/enums?

Reply:
yeah, why not

Reply:
Heh, if both android and ios gcloud config has common `device` option so probably it should be in CommonGcloudConfig.